experanto/datasets.py

```diff
@@ -438,11 +447,219 @@ def __getitem__(self, idx) -> dict:
+                                 else x)
+                transform_list = [v for v in self.modality_config.screen.transforms.values() if isinstance(v, torch.nn.Module)]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                transform_list = [ToTensor()]
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                transform_list.append(
+                    torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])
+                )
+            transforms[device_name] = Compose(transform_list)
+        return transforms
+
+    def get_swap_time_valid_conditions(self) -> np.ndarray:
+        """
+        Determines valid swap times based on the meta conditions in the modality configuration.
+        """
+        valid_conditions = self.modality_config["screen"]["valid_condition"]
+        include_blanks = self.modality_config["screen"]["include_blanks"]
+
+        valid_indices = np.ones(len(self.meta_conditions["valid_trial"]), dtype=bool)
+
+        for condition_key, condition_value in valid_conditions.items():
+            condition_array = np.array(self.meta_conditions[condition_key])
+            valid_indices &= (condition_array == condition_value) | (
+                (condition_array == "blank") & include_blanks
+            )
+        self._valid_indices = valid_indices
+        # Return the swap times that satisfy all conditions
+        return self._experiment.devices['screen'].timestamps[valid_indices]
+
+
+    def shuffle_valid_screen_times(self) -> None:
+        times = self._full_valid_sample_times
+        self._valid_screen_times = np.sort(np.random.choice(times, size=len(times) // self.sample_stride, replace=False))
+    
+
+    @staticmethod
+    def get_batch_aligned_split(n_images, train_frac, batch_size, seed):
+        if seed is not None:
+            np.random.seed(seed)
+
+        indices = np.arange(n_images)
+        np.random.shuffle(indices)
+
+        # Compute training size and align up to the next batch
+        train_size = int(n_images * train_frac)
+        train_size = ((train_size + batch_size - 1) // batch_size) * batch_size  # Align up
+        train_size = min(train_size, n_images)  # Ensure it doesn't exceed total images
+
+        train_indices = indices[:train_size]
+        val_indices = indices[train_size:]
+        return train_indices, val_indices
+
+    def split(self, train_frac=0.8, seed=None):
+        """
+        Split the dataset into train and validation subsets using the same logic as the static loader.
+        """
+        # Get train and validation indices using the static loader's logic
+        train_indices, val_indices = get_validation_split(len(self), train_frac, seed)
```

Comment:
I think that
1) there should be train-validation-test splits and not just 2 
2) these tier labels should be fixed somewhere in metadata, such that the splits are comparable between different runs with different seeds (which could be useful to test stability with different data shuffling order and model initialization) and also comparable between different ends, etc as splits might be different even with the same seed in different env